aws/sdk/examples/s3/src/bin/list-objects.rs

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use std::process;
+
+use s3::{Client, Config, Region};
+
+use aws_types::region::ProvideRegion;
+
+use structopt::StructOpt;
+use tracing_subscriber::fmt::format::FmtSpan;
+use tracing_subscriber::fmt::SubscriberBuilder;
+
+#[derive(Debug, StructOpt)]
+struct Opt {
+    /// The default region
+    #[structopt(short, long)]
+    default_region: Option<String>,
+
+    /// The name of the bucket
+    #[structopt(short, long)]
+    bucket: String,
+
+    /// Whether to display additional information
+    #[structopt(short, long)]
+    verbose: bool,
+}
+
+/// Lists the objects in an Amazon S3 bucket.
+/// # Arguments
+///
+/// * `-n NAME` - The name of the bucket.
+/// * `[-d DEFAULT-REGION]` - The region containing the bucket.
+///   If not supplied, uses the value of the **AWS_DEFAULT_REGION** environment variable.
+///   If the environment variable is not set, defaults to **us-west-2**.
+/// * `[-v]` - Whether to display additional information.
+#[tokio::main]
+async fn main() {
+    let Opt {
+        default_region,
+        bucket,
+        verbose,
+    } = Opt::from_args();
+
+    let region = default_region
+        .as_ref()
+        .map(|region| Region::new(region.clone()))
+        .or_else(|| aws_types::region::default_provider().region())
+        .unwrap_or_else(|| Region::new("us-west-2"));
+
+    if verbose {
+        println!("S3 client version: {}", s3::PKG_VERSION);
+        println!("Region:            {:?}", &region);
+
+        SubscriberBuilder::default()
+            .with_env_filter("info")
+            .with_span_events(FmtSpan::CLOSE)
+            .init();
+    }
+
+    let config = Config::builder().region(&region).build();
+
+    let client = Client::from_conf(config);
+
+    match client.list_objects().bucket(&bucket).send().await {
+        Ok(resp) => {
+            match resp.contents {
+                None => {}
+                Some(objects) => {
+                    // vector of Object items
+                    println!("Objects:");
+                    for (_, x) in objects.iter().enumerate() {
```

Comment:
I'm a little confused here, why get an index with `.enumerate()` but then just ignore it?

What about something like:

```rust
println!("Objects:");
for object in resp.contents.unwrap_or_default() {
   println!(" {}", object.key.expect("objects have keys"));
}
```

Reply:
Rookie mistake. Fixed.